### PR TITLE
'istioctl proxy-status' and 'version' to use XDS version

### LINKED ISSUE #27324: 'istioctl proxy-status' and 'version' to use XDS version, not legacy version
This work item is replace the implementation of 'proxy-status' and 'version' with the experimental counterparts.

With this change istioctl stops call calling /version and /debug for those commands, reducing our dependency on the old debug endpoints.

Should we take votes if we should keep the old versions around, hidden and deprecated, or get rid of them completely.  I am in favor of getting rid of them completely.

**This work item is easy, but there are implications.  What if the XDS endpoint is unreachable but the older debug endpoint is reachable?  We had trouble with 1.7.x, switching from `exec` to `port-forward` to improve security, and running into problems with users who didn't have port-forwarding capability.

### PR DESCRIPTION
**Pull request by moricho**
istioctl: replace the implementation of 'proxy-status' and 'version' with the experimental counterparts

Fixes #27324 

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: istioctl/cmd/version.go [hunk 1]
```diff
@@ -29,39 +29,12 @@ import (
...
-			err := flag.Value.Set("true")
-			if err != nil {
-				fmt.Fprintf(os.Stdout, "set flag %q as true failed due to error %v", flag.Name, err)
-			}
-		}
-	})
-	return versionCmd
-}
-
 func getRemoteInfo(opts clioptions.ControlPlaneOptions) (*istioVersion.MeshInfo, error) {
```
Comment (shamsher31) [inline istioctl/cmd/version.go:38]: Can you also remove this function if it is not useful? Also I think following funcs are not required
`getRemoteInfoWrapper`
`getProxyInfoWrapper`

Comment (shamsher31) [reply istioctl/cmd/version.go:38]: I think linter caught it https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/27387/lint_istio/19992#1:build-log.txt%3A13

**Review by shamsher31 — COMMENTED** [review]
You also need to update the test to match new output for `proxy-status`.

**Comment by esnible** [comment]
I am adding `hold` on this not because it has any problem, but because `istioctl x version` didn't work for me this morning.  I don't want this to merge until I understand why not.

File: istioctl/cmd/version.go (hunk 1 above)
Comment (moricho) [reply istioctl/cmd/version.go:38]: fixed!

**Comment by moricho** [comment]
@shamsher31 @esnible 
I'm now working on fixing failed unit tests, and I'd like you to get your opinion.

For querying the `status` and `version` from an XDS server, an XDS response should be mocked during testing.
However, it is difficult to mock it because a dial connection to the ADS server is generated in the `xds.GetXdsResponse` function.
Hence I think it's better to implement an XDS client that has a dial connection to the ADS server and that implement the `GetXdsResponse` method. That makes it easier for us to mock the XDS response.

What do you think about this? If it's worth it, I'll implement the client on this(or another) PR.

**Comment by esnible** [comment]
> What do you think about this? If it's worth it, I'll implement the client on this(or another) PR.

I have been nervous about mocking XDS responses.  We had much trouble in 2019 with Envoy config mocks that became stale as Pilot generated different configs.  Unit tests have the same problem.

It makes sense to either
- Create a mock framework sufficiently powerful enough to get Pilot to cough up an XDS response for the mock
- Do 100% of the testing in integration tests, skipping unit tests entirely.

Unit tests for these commands give a false sense of security if they are based on checked-in canned XDS responses.

**Pull request closed without merging**